gems/aws-sdk-core/spec/aws/plugins/sign_spec.rb

```diff
@@ -202,6 +202,16 @@ def call(context)
               to eq (now.utc + 1000).strftime("%Y%m%dT%H%M%SZ")
           end
         end
+
+        it 'Adds metrics from the credential providers' do
```

Comment:
This might not test what you expect. It is just testing that any of the metrics have the letter "e"? But what if we add metric "ae"? I think you just want to expect() that after you call operation, that creds.metrics is called, or if you want to do a behavioral test, you should properly parse out the metrics list like you do in the other tests.